sqlmesh/core/analytics/collector.py

```diff
@@ -0,0 +1,351 @@
+            command_name: The name of the command.
+            command_args: The arguments of the command.
+            parent_command_names: The names of the parent commands.
+        """
+        self._on_command(
+            "CLI_COMMAND", command_name, command_args, parent_command_names=parent_command_names
+        )
+
+    def on_magic_command(self, *, command_name: str, command_args: t.Collection[str]) -> None:
+        """Called when a Notebook magic command is executed.
+
+        Args:
+            command_name: The name of the command.
+            command_args: The arguments of the command.
+        """
+        self._on_command("MAGIC_COMMAND", command_name, command_args)
+
+    def on_python_api_command(self, *, command_name: str, command_args: t.Collection[str]) -> None:
+        """Called when a Python method is called directly.
+
+        Args:
+            command_name: The name of the command.
+            command_args: The arguments of the command.
+        """
+        self._on_command("PYTHON_API_COMMAND", command_name, command_args)
+
+    def on_project_loaded(
+        self,
+        *,
+        project_type: str,
+        models_count: int,
+        audits_count: int,
+        standalone_audits_count: int,
+        macros_count: int,
+        jinja_macros_count: int,
+        load_time_sec: float,
+        state_sync_fingerprint: str,
+        project_name: str,
+    ) -> None:
+        """Called when a project is loaded.
+
+        Args:
+            project_type: The type of the project. Eg. "dbt" or "native".
+            models_count: The number of models in the project.
+            audits_count: The number of audits in the project.
+            standalone_audits_count: The number of standalone audits in the project.
+            macros_count: The number of macros in the project.
+            jinja_macros_count: The number of Jinja macros in the project.
+            load_time_sec: The time it took to load the project in (fractional) seconds.
+            state_sync_fingerprint: The fingerprint of the state sync configuration.
+            project_name: The name of the project.
+        """
+        dbt_version = None
+        project_type = project_type.upper()
+        if project_type == "DBT":
+            from dbt import version
+
+            dbt_version = version.__version__
+
+        self._add_event(
+            "PROJECT_LOADED",
+            {
+                "project_type": project_type.upper(),
+                "models_count": models_count,
+                "audits_count": audits_count,
+                "standalone_audits_count": standalone_audits_count,
+                "macros_count": macros_count,
+                "jinja_macros_count": jinja_macros_count,
+                "load_time_ms": int(load_time_sec * 1000),
+                "state_sync_fingerprint": state_sync_fingerprint,
+                "project_name_hash": _anonymize(project_name),
+                "dbt_version": dbt_version,
```

Comment:
i think dbt_version should probably only be in the dict if it exists. 

you could create the dict above the dbt check, and then set the dict inside the dbt if